WalletWasabi.Gui/Controls/WalletExplorer/CoinListViewModel.cs

```diff
@@ -385,7 +393,20 @@ public CoinListViewModel(Global global, CoinListContainerType coinListContainerT
 					}
 				});
 
-			InitList = ReactiveCommand.Create(() => OnOpen(), outputScheduler: RxApp.MainThreadScheduler);
+			InitList = ReactiveCommand.CreateFromTask(async () =>
+			{
+				try
+				{
+					IsCoinListLoading = true;
+					// We have to wait for the UI to became visible to the user.
+					await Task.Delay(800); // Let other tasks run to display the gui.
+					OnOpen();
+				}
+				finally
+				{
+					IsCoinListLoading = false;
+				}
+			}, outputScheduler: RxApp.MainThreadScheduler);
```

Comment:
Can be removed. I mistakenly thought it does something useful back then.

Reply:
It is as useful as the `.ObserveOn(RxApp.MainThreadScheduler)` IMO. If you are call `ReactiveCommand.Execute()` from another thread Rx will synchronize. We are just not a calling the commands from another thread.

Reply:
Removed it because of code conventions.

Reply:
No. It doesn't force the command to run on the `MainThreadScheduler`, it does something else.